packages/flutter/lib/src/widgets/mouse_detector.dart

```diff
@@ -0,0 +1,176 @@
+///  Copyright 2018 The Chromium Authors. All rights reserved.
+///  Use of this source code is governed by a BSD-style license that can be
+///  found in the LICENSE file.
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/gestures.dart';
+import 'package:flutter/rendering.dart';
+
+import 'annotated_region.dart';
+import 'framework.dart';
+
+/// A widget that detects changes in mouse pointer positions when a mouse button
+/// is not pressed. Note that this only applies to mouse pointers: touch
+/// pointers cannot be detected unless their "button" is down (which is a fancy
+/// way of saying that we don't know where your fingers are when they're not on
+/// the screen).
```

Comment:
see comments earlier about the purported existence of hardware that does support this

Reply:
Yeah, I agree, such hardware may exist, but this is not the paradigm that the Engine defines: touch events don't report movement if they aren't down, because their associated pointers don't exist if they're not down. I removed the parenthetical, since that's a statement that is broader than the engine definition of touch move events.

We should probably change the definition in the engine, but that can't really happen until we know exactly what form the devices take that we want to support (e.g. do they support an `altitude`, `pointerAngle`, etc., or just their own definition of "hover").